src/NLog.Targets.Redis/RedisTarget.cs

```diff
@@ -50,42 +36,57 @@ public string Key
         /// <summary>
         /// Sets the database id to be used in redis if the log entries are sent to a list. Defaults to 0
         /// </summary>
-        public int Db { get; set; }
+        public Layout Db { get; set; }
 
         /// <summary>
         /// Sets the password to be used when accessing Redis with authentication required
         /// </summary>
-        public string Password { get; set; }
+        public Layout Password { get; set; }
 
         private RedisConnectionManager _redisConnectionManager;
 
-        internal virtual RedisConnectionManager CreateConnectionManager()
+        internal virtual RedisConnectionManager CreateConnectionManager(string host, int port, int db, string password)
         {
-            return new RedisConnectionManager(Host, Port, Db, Password);
+            return new RedisConnectionManager(host, port, db, password);
         }
 
         protected override void InitializeTarget()
         {
-            _redisConnectionManager = CreateConnectionManager();
+            var host = Host?.Render(LogEventInfo.CreateNullEvent());
+            var password = Password?.Render(LogEventInfo.CreateNullEvent());
+
+            var renderedPort = Port.Render(LogEventInfo.CreateNullEvent());
+            if (!int.TryParse(renderedPort, out var port))
+            {
+                throw new Exception($"Unable to parse Port:{renderedPort}");
+            }
+            var db = 0;
+            if (Db != null)
+            {
+                var renderedDb = Db.Render(LogEventInfo.CreateNullEvent());
+                if (!int.TryParse(renderedDb, out db))
+                {
+                    InternalLogger.Warn("RedisTarget: Unable to parse Db:{0}", renderedDb);
```

Comment:
Maybe specify that we are defaulting to "0" when the parse failed?

{nameof(RedisTarget)}: Unable to parse db as a number: {renderedDb} - falling back to the default db 0.

Reply:
IMO it's always a good idea to explain (in short) what we are doing when we continue with a warning. So I vote for yes.